nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/Environment.java

```diff
@@ -109,13 +110,29 @@ public Environment(boolean ignoreEnvironment) {
         }
         if (fname != null) {
             File f = new File(fname);
-
             try {
-                if ((f.exists() || f.createNewFile()) && f.isFile() && f.canWrite()) {
+                if (f.isFile() && f.canWrite()) {
+                    logfile = f;
+                } else {
+                    File dir = f.getParentFile();
+                    if (!dir.isDirectory() && !dir.mkdirs()) {
+                        throw new IllegalStateException("Could not create a log file: " + f);
```

Comment:
Maybe it's better adding a reason to the exception message? Something like "log directory not found"?